### PR TITLE
Fix indexing behavior in NumpyTupleDataset

### PR DESCRIPTION
**Pull request by mottodora**
This PR contains non compatibility change.

### CODE REVIEW

File: chainer_chemistry/datasets/numpy_tuple_dataset.py [hunk 1]
```diff
@@ -35,7 +35,8 @@ def __init__(self, *datasets):
 
     def __getitem__(self, index):
         batches = [dataset[index] for dataset in self._datasets]
-        if isinstance(index, slice):
+        if isinstance(index, slice) or isinstance(index, list) or\
```
Comment (corochann) [inline chainer_chemistry/datasets/numpy_tuple_dataset.py]: you may write
`if isinstance(index, (slice, list, numpy.ndarray)):`

File: tests/datasets_tests/test_numpy_tuple_dataset.py [hunk 2]
```diff
@@ -47,6 +55,38 @@ def test_get_item_slice_index(self, data, index):
             for a, e in six.moves.zip(tuple_a, tuple_e):
                 numpy.testing.assert_array_equal(a, e)
 
+    @pytest.mark.parametrize('index', [numpy.asarray([2, 0])])
```
Comment (corochann) [inline tests/datasets_tests/test_numpy_tuple_dataset.py]: Can you add array with length 1?

File: tests/datasets_tests/test_numpy_tuple_dataset.py [hunk 3]
```diff
@@ -47,6 +55,38 @@ def test_get_item_slice_index(self, data, index):
...
+        expect = [tuple([batch[i] for batch in batches])
+                  for i in six.moves.range(length)]
+
+        assert len(actual) == len(expect)
+        for tuple_a, tuple_e in six.moves.zip(actual, expect):
+            assert len(tuple_a) == len(tuple_e)
+            for a, e in six.moves.zip(tuple_a, tuple_e):
+                numpy.testing.assert_array_equal(a, e)
+
+    @pytest.mark.parametrize('index', [[2, 0]])
```
Comment (corochann) [inline tests/datasets_tests/test_numpy_tuple_dataset.py]: Can you add list with length 1?

**Comment by mottodora** [comment]
Fix.

**Pull request merged**